pkg/controller/controller_binding.go

```diff
@@ -119,13 +122,16 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 			binding.Spec.InstanceRef.Name,
 		)
 		glog.Info(s)
-		c.updateBindingCondition(
+		if err := c.updateBindingCondition(
 			binding,
 			v1alpha1.BindingConditionReady,
 			v1alpha1.ConditionFalse,
 			errorWithOngoingAsyncOperation,
 			errorWithOngoingAsyncOperationMessage,
-		)
+		); err != nil {
+			glog.Errorf("updating binding condition (%s)", err)
+			return err
+		}
 		c.recorder.Event(binding, api.EventTypeWarning, errorWithOngoingAsyncOperation, s)
```

Comment:
@MHBauer I left them where they were before. I think it's the right behavior to not emit an event if the update failed, because the state didn't change. Thoughts?